### DISCUSSION TITLE
Jest / Testing-Library: ReferenceError: MutationObserver is not defined

### DISCUSSION THREAD

**Discussion by frankbirzle** [react-hook-form/react-hook-form#3197] [Q&A] [answered]
Hi there,

I know there've been a few posts regarding this problem, however I can't get it to work (create-react-app project with template typescript).

When trying to test a component that uses useForm I get the following error:
`ReferenceError: MutationObserver is not defined`
My src/setupTests.ts looks like this:
```
import "@testing-library/jest-dom";
```
Any input is highly appreciated.
Thx!

Edit: formatting

**Comment by bluebill1049** [comment]
strange... as you can see this lib use that package for testing as well. any chance to share the github repo?

**Reply by fadi-george** [reply]
Isn't `import 'mutationobserver-shim';` required?

**Reply by frankbirzle** [reply]
Not anymore, this has been replace by `import "@testing-library/jest-dom";`

**Comment by bluebill1049** [comment, marked as answer]
Is this still an issue for you?

**Reply by frankbirzle** [reply]
Sorry for only getting back to you now. 

I updated the entire project to CRA 4.0 and also to react 17.0.1 and now everything works as expected. Thx!
